Retain cursor focus on user-mentions block when the popover opens

glenpadua opened the pull request:
Typing "@" in a textarea wrapped with the withUserMentions HOC would open a popover containing a
user suggestion list that would cause the textarea to lose focus, requiring the user to manually
place the cursor back to continue default behavior. Adding a default prop isFocusEnabled which
defaults to true to check if the deferred focusPopover method should be called.

#### Changes proposed in this Pull Request

* Add default prop isFocusEnabled
* Add check before calling focusPopover
* Default isFocusEnabled for Popover to true
* Pass isFocusEnabled as false to the popover in the UserMentions component

#### Testing instructions

* Visit http://calypso.localhost:3000/devdocs/blocks/user-mentions
* Press '@' and start typing for user matches
* User suggestions show up and textarea remains in focus, allowing the user to perform keyboard operations as expected.

Fixes #37625

glenpadua commented:
I was unable to test the fix where the issue was originally reported as I'm not sure where it shows up 🤔 

> Internal users only: browse to a P2 post in Reader (full post, e.g. /read/feeds/:feed_id/posts/:post_id)

bluefuton commented:
Thanks for tackling this @glenpadua! As this component is used throughout Calypso, I think we have to be cautious about removing this. There may be other places using `Popover` without `PopoverMenu` that rely on this behaviour.

Can we try adding a prop that is provided to the `Popover` component to turn off the focus instead, that defaults to enabled?

glenpadua commented:
@bluefuton That sounds fair. Will do that then!

glenpadua committed:
Retain cursor focus on user-mentions block when the popover opens

Typing "@" in a textarea wrapped with the withUserMentions HOC would open a popover containing a
user suggestion list that would cause the textarea to lose focus, requiring the user to manually
place the cursor back to continue default behaviour. Adding a default prop isFocusEnabled which
defaults to true to check if the deferred focusPopover method should be called.

* Add default prop isFocusEnabled
* Add check before calling focusPopover
* Default isFocusEnabled for Popover to true
* Pass isFocusEnabled as false to the popover in the UserMentions component

glenpadua commented:
@bluefuton I've added a default prop called `isFocusEnabled`. Let me know if you have a better name for it and if I need to make any more changes!

bluefuton commented on client/components/popover/index.jsx:
```diff
@@ -38,6 +38,7 @@ class PopoverInner extends Component {
 		className: '',
 		closeOnEsc: true,
 		isRtl: false,
+		isFocusEnabled: false,
```
As the popover is currently always focussed, can we default this to `true`, then pass in `isFocusEnabled={ false }` for the user mentions popover?

glenpadua replied:
@bluefuton Have made the change! 👍

bluefuton reviewed (approved):
Looks good to me. Thanks for fixing this @glenpadua! 🚢
